modules/images/webp-uploads/load.php

```diff
@@ -97,7 +97,10 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 			continue;
 		}
 
-		if ( webp_uploads_should_discard_additional_image_file( $metadata, $image ) ) {
+		if (
+			! empty( $metadata['sources'][ $mime_type ] ) &&
+			webp_uploads_should_discard_additional_image_file( $metadata['sources'][ $mime_type ], $image )
```

Comment:
Oh wow, that's a critical fix I guess? The `$metadata` above looks completely wrong, since it's the full attachment metadata right? Guess we never found this before.

Reply:
Yeah, i was also surprised to see that we use the whole metadata array here whilst we needed only source information for the current mime type.

Reply:
Actually, passing `$metadata` here is not wrong. What we expect inside of this function is an array that has keys `file` and `filesize`. The overall attachment metadata _should_ have that field, at least since WordPress 6.0 (see e.g. core's `wp_generate_attachment_metadata()` and `_wp_image_meta_replace_original()`). Did you potentially run into this problem with an older image, our in a version below 6.0?

It would also be okay passing the individual source here, but technically they should have the same data, so we really need to figure out what's going wrong here. Could you follow up on that?

Side note, maybe we should raise the minimum required WordPress version to 6.0 soon - it would help this one, and also 6.1 is coming out soon, and our policy is to support at least the latest 2 versions.

Reply:
@felixarntz, yes, we have `file` and `filesize` properties with WordPress 6.0, I was on an older version. I'll revert this change.

> Side note, maybe we should raise the minimum required WordPress version to 6.0 soon - it would help this one, and also 6.1 is coming out soon, and our policy is to support at least the latest 2 versions.

Yes, I think we should definitely do it if we rely on a certain functionality from WP 6.0 in our codebase.

Reply:
Ok, reverted this change.